File: Example/Objective-C/KGViewSeparatorsExample/UITableViewCell+Separators.m
//
//  UITableViewCell+Separators.m
//  KGViewSeparators
//
//  Created by Krisjanis Gaidis on 12/27/15.
//  Copyright © 2015 KG. All rights reserved.
//

#import "UITableViewCell+Separators.h"
#import "UIView+KGViewSeparators.h"
#import "KGViewSeparatorConfiguration+Defaults.h"

@implementation UITableViewCell (Separators)

- (void)showRedTopSeparator:(BOOL)show {
    [self.contentView kg_show:show separator:KGViewSeparatorTop color:[UIColor redColor] lineWidth:1.0 insets:UIEdgeInsetsZero];
}

- (void)showRedBottomSeparator:(BOOL)show {
    [self.contentView kg_show:show separator:KGViewSeparatorBottom color:[UIColor redColor] lineWidth:1.0 insets:UIEdgeInsetsZero];
}

- (void)showGreenTopSeparator:(BOOL)show {
    [self.contentView kg_show:show separator:KGViewSeparatorTop color:[UIColor greenColor] lineWidth:[KGViewSeparatorConfiguration defaultLineWidth] insets:UIEdgeInsetsMake(0, 15, 0, 15)];
}

- (void)showGreenBottomSeparator:(BOOL)show {
    [self.contentView kg_show:show separator:KGViewSeparatorBottom color:[UIColor greenColor] lineWidth:[KGViewSeparatorConfiguration defaultLineWidth] insets:UIEdgeInsetsMake(0, 15, 0, 15)];
}

- (void)showBlackTopSeparator:(BOOL)show {
    [self.contentView kg_show:show separator:KGViewSeparatorTop color:[UIColor blackColor] lineWidth:2.0 insets:UIEdgeInsetsMake(0, 0, 0, 40)];
}

- (void)showBlackBottomSeparator:(BOOL)show {
    [self.contentView kg_show:show separator:KGViewSeparatorBottom color:[UIColor blackColor] lineWidth:2.0 insets:UIEdgeInsetsMake(0, 0, 0, 40)];
}

@end

File: Example/Swift/KGViewSeparatorsExample/UITableViewCell+Separators.swift
//
//  UITableViewCell+Separators.swift
//  KGViewSeparatorsExample
//
//  Created by Krisjanis Gaidis on 12/28/15.
//  Copyright © 2015 KG. All rights reserved.
//

import Foundation

extension UITableViewCell {
    func showTopSeparator(show: Bool) {
        self.contentView.kg_show(show, separator: .Top, color:  UIColor.blackColor(), lineWidth: (1.0 / UIScreen.mainScreen().scale), insets: UIEdgeInsetsZero)
    }
    
    func showBottomSeparator(show: Bool) {
        self.contentView.kg_show(show, separator: .Bottom, color:  UIColor.blackColor(), lineWidth: (1.0 / UIScreen.mainScreen().scale), insets: UIEdgeInsetsZero)
    }
}